redirector/go/main.go

```diff
@@ -0,0 +1,266 @@
+			continue
+		}
+		fuseMountPoints = append(fuseMountPoints, v.Path)
+	}
+
+	if len(fuseMountPoints) == 0 {
+		return "", fuse.ENOENT
+	}
+	if len(fuseMountPoints) == 1 {
+		return fuseMountPoints[0], nil
+	}
+
+	// If there is more than one, pick the first one alphabetically
+	// that has "keybase" in the path.
+	sort.Strings(fuseMountPoints)
+	for _, mp := range fuseMountPoints {
+		// TODO: a better regexp that will rule out keybase.staging if
+		// we're in prod mode, etc.
+		if !strings.Contains(mp, "keybase") {
+			continue
+		}
+		// Double-check that this is a real KBFS mount.
+		if _, err := os.Stat(filepath.Join(mp, ".kbfs_error")); err != nil {
+			continue
+		}
+		return mp, nil
+	}
+
+	// Give up and return the first one.
+	return fuseMountPoints[0], nil
+}
+
+func (r *root) ReadDirAll(ctx context.Context) ([]fuse.Dirent, error) {
+	_, err := r.findKBFSMount(ctx)
+	if err != nil {
+		if err == fuse.ENOENT {
+			return []fuse.Dirent{}, nil
+		}
+		return []fuse.Dirent{}, err
+	}
+
+	// TODO: show the `kbfs.error.txt" and "kbfs.nologin.txt" files if
+	// they exist?  As root, it is hard to figure out if they're
+	// there, though.
+	return []fuse.Dirent{
+		{
+			Type: fuse.DT_Link,
+			Name: "private",
+		},
+		{
+			Type: fuse.DT_Link,
+			Name: "public",
+		},
+		fuse.Dirent{
+			Type: fuse.DT_Link,
+			Name: "team",
+		},
+	}, nil
+}
+
+func (r *root) Lookup(
+	ctx context.Context, req *fuse.LookupRequest, resp *fuse.LookupResponse) (
+	n fs.Node, err error) {
+	mountpoint, err := r.findKBFSMount(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	resp.EntryValid = 0
+	switch req.Name {
+	case "private", "public", "team", ".kbfs_error", ".kbfs_metrics",
+		".kbfs_profiles", ".kbfs_reset_caches", ".kbfs_status",
+		"kbfs.error.txt", "kbfs.nologin.txt", ".kbfs_enable_auto_journals",
+		".kbfs_disable_auto_journals", ".kbfs_enable_block_prefetching",
+		".kbfs_disable_block_prefetching", ".kbfs_enable_debug_server",
+		".kbfs_disable_debug_server":
+		return symlink{filepath.Join(mountpoint, req.Name)}, nil
+	}
+	return nil, fuse.ENOENT
+}
+
+func main() {
+	currUser, err := user.Current()
+	if err != nil {
+		panic(err)
+	}
+	if currUser.Uid != "0" {
+		runtime.LockOSThread()
+		_, _, errNo := syscall.Syscall(syscall.SYS_SETUID, 0, 0, 0)
+		if errNo != 0 {
+			fmt.Fprintf(os.Stderr, "Can't setuid: %+v\n", errNo)
+			os.Exit(1)
+		}
+	}
+
+	options := []fuse.MountOption{fuse.AllowOther()}
+	options = append(options, fuse.FSName("keybase-redirector"))
+	options = append(options, fuse.ReadOnly())
+	if runtime.GOOS == "darwin" {
+		options = append(options, fuse.OSXFUSELocations(kbfusePath))
+		options = append(options, fuse.VolumeName("keybase-redirector"))
+		options = append(options, fuse.NoBrowse())
+	}
+
+	c, err := fuse.Mount(os.Args[1], options...)
+	if err != nil {
+		fmt.Printf("Mount error, exiting cleanly: %+v\n", err)
+		os.Exit(0)
+	}
+
+	interruptChan := make(chan os.Signal, 1)
+	signal.Notify(interruptChan, os.Interrupt)
```

Comment:
Can we do this on a `SIGTERM` as well?

Reply:
SIGTERM is actually probably more often since that's the default for `kill`/`killall`